pinot-integration-test-base/src/test/java/org/apache/pinot/integration/tests/BaseClusterIntegrationTest.java

```diff
@@ -637,12 +637,26 @@ protected boolean injectTombstones() {
     return false;
   }
 
-  protected void createAndUploadSegmentFromFile(TableConfig tableConfig, Schema schema, String dataFilePath,
+  protected void createAndUploadSegmentFromClasspath(TableConfig tableConfig, Schema schema, String dataFilePath,
       FileFormat fileFormat, long expectedNoOfDocs, long timeoutMs) throws Exception {
     URL dataPathUrl = getClass().getClassLoader().getResource(dataFilePath);
     assert dataPathUrl != null;
     File file = new File(dataPathUrl.getFile());
 
+    createAndUploadSegmentFromFile(tableConfig, schema, file, fileFormat, expectedNoOfDocs, timeoutMs);
+  }
+
+  /// @deprecated use createAndUploadSegmentFromClasspath instead, given what this class does is to look for
+  /// dataFilePath on the classpath
+  @Deprecated
+  protected void createAndUploadSegmentFromFile(TableConfig tableConfig, Schema schema, String dataFilePath,
+      FileFormat fileFormat, long expectedNoOfDocs, long timeoutMs) throws Exception {
+    createAndUploadSegmentFromClasspath(tableConfig, schema, dataFilePath, fileFormat, expectedNoOfDocs, timeoutMs);
+  }
+
+  protected void createAndUploadSegmentFromFile(TableConfig tableConfig, Schema schema, File file,
```

Comment:
I need this new method to upload files that are created on the fly instead of being stored in the classpath as static examples